### PR TITLE
Added HQ support automatically

### PR DESCRIPTION
**Pull request by Looty**
changed the code
changed README

### CODE REVIEW

File: Plugin.cs [hunk 1]
```diff
@@ -153,15 +148,71 @@ private void OnNetworkEvent(IntPtr dataPtr, ushort opCode, uint sourceActorId, u
             if (!this.configuration.alwaysOn && (!this.configuration.smart || !Retainer())) return;
             var listing = MarketBoardCurrentOfferings.Read(dataPtr);
             var i = 0;
-            if (this.configuration.hq && this.items.Single(j => j.RowId == listing.ItemListings[0].CatalogId).CanBeHq)
-            {
```
Comment (tesu) [inline Plugin.cs:157]: This should be the only line that uses this.items, so by removing it you should be able to get rid of the Lumina.Excel.GeneratedSheets dependency.

File: README.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ The usual workflow for this is the following:
 6. Update your item's price with the number you noted
 7. Confirm your new price
 
-This plugin eliminates the mental notekeeping steps in the middle by automatically copying the number you want onto your clipboard when you open the current listings window.
+This plugin eliminates the mental notekeeping steps in the middle by automatically copying the number you want onto your clipboard when you open the current listings window (suppots HQ items aswell).
```
Comment (tesu) [inline README.md:15]: s/suppots/supports/ and s/aswell/as well/. I would just reword this as 
`This plugin eliminates the mental notekeeping steps in the middle by automatically copying the number you want (with automatic HQ support) onto your clipboard when you open the current listings window.`

File: Plugin.cs [hunk 3]
```diff
@@ -153,15 +148,71 @@ private void OnNetworkEvent(IntPtr dataPtr, ushort opCode, uint sourceActorId, u
...
+
+            // Load addon ptr from memory
+            var toolTipPtr = pi.Framework.Gui.GetUiObjectByName("ItemDetail", 1) + 0x258;
```
Comment (tesu) [inline Plugin.cs:153]: Is this `0x258` fixed? It's fine if it changes a few times per expac; I just don't want to have to push a new release every patch/hotfix if it changes all the time.

File: Plugin.cs [hunk 4]
```diff
@@ -153,15 +148,71 @@ private void OnNetworkEvent(IntPtr dataPtr, ushort opCode, uint sourceActorId, u
...
+                isCurrentItemHQ = true;
+            else
+                isCurrentItemHQ = false;
+
+            if (isCurrentItemHQ) {
                 while (i < listing.ItemListings.Count && !listing.ItemListings[i].IsHq) i++;
                 if (i == listing.ItemListings.Count) return;
+            } else {
+                while (i < listing.ItemListings.Count && listing.ItemListings[i].IsHq) i++;
+                if (i == listing.ItemListings.Count) return;
```
Comment (tesu) [inline Plugin.cs:168]: What's going on here? If `isCurrentItemHQ == false` we should just proceed with `i = 0`, right? I don't see why your code needs to change any of the logic after deciding whether to get lowest vs lowest HQ.

File: Plugin.cs [hunk 5]
```diff
@@ -153,15 +148,71 @@ private void OnNetworkEvent(IntPtr dataPtr, ushort opCode, uint sourceActorId, u
...
+            // Load addon ptr from memory
+            var toolTipPtr = pi.Framework.Gui.GetUiObjectByName("ItemDetail", 1) + 0x258;
+            var toolTipItemName = GetSeStringText(GetSeString(toolTipPtr));
+            bool isCurrentItemHQ = false;
+
+            // Checks for HQ symbol in item name
+            if (toolTipItemName.Substring(toolTipItemName.Length - 1) == "")
+                isCurrentItemHQ = true;
+            else
+                isCurrentItemHQ = false;
```
Comment (tesu) [inline Plugin.cs:161]: Just make this `bool isCurrentItemHQ = toolTipItemName.Substring(toolTipItemName.Length - 1) == ""`

File: Configuration.cs [hunk 6]
```diff
@@ -11,7 +11,6 @@ public class Configuration : IPluginConfiguration
 
         public bool alwaysOn { get; set; } = false;
         public int delta { get; set; } = 1;
-        public bool hq { get; set; } = false;
```
Comment (tesu) [inline Configuration.cs:14]: Curious if this is safe to do without bumping the config version and doing something more careful. Did you old config work fine after making this change (minus the hq setting obviously?)

File: Plugin.cs [hunk 7]
```diff
@@ -153,15 +148,71 @@ private void OnNetworkEvent(IntPtr dataPtr, ushort opCode, uint sourceActorId, u
...
+        }
+
+        /// <summary>
+        /// Parse bytes to SeString
+        /// </summary>
+        /// <param name="bytes"></param>
+        /// <returns></returns>
+        private SeString GetSeString(byte[] bytes) {
+            return pi.SeStringManager.Parse(bytes);
+        }
```
Comment (tesu) [inline Plugin.cs:216]: Is this copied from somewhere? Not a problem if it is, just curious what the original source is.

**Review by tesu — CHANGES_REQUESTED** [review]

File: Plugin.cs (hunk 1 above)
Comment (Looty) [reply Plugin.cs:157]: Wasn't sure what this.items refers to..
Does this refers to the item we want to sell?

File: README.md (hunk 2 above)
Comment (Looty) [reply README.md:15]: My typo there
Sure

File: Plugin.cs (hunk 1 above)
Comment (tesu) [reply Plugin.cs:157]: I mean the `this.items.Single(j => j.RowId == listing.ItemListings[0].CatalogId).CanBeHq)` is the only time we touch `this.items`, so you can get rid of the Excel sheet if we're not using it anymore.

File: Plugin.cs (hunk 3 above)
Comment (Looty) [reply Plugin.cs:153]: Fixed - yes
That's the address I found in ReClass
I agree - it would be a hassle to maintain it

I tested this a few times a day for the last 2-3 days and its still working
But of course - needed to be furtherly tested

File: Plugin.cs (hunk 4 above)
Comment (Looty) [reply Plugin.cs:168]: If it is false - meaning the current item we're looking it is not HQ
As far as I understand, the i refers to the index of the listing

The only thing I changed this part was the !(boolean) before the listing.ItemListings[i].IsHq depends on the quality
I left the logic the same

File: Plugin.cs (hunk 5 above)
Comment (Looty) [reply Plugin.cs:161]: Got it

File: Configuration.cs (hunk 6 above)
Comment (Looty) [reply Configuration.cs:14]: Not sure if I understand the meaning of "bumping the config version"
After removing this leftover - everything was still working

File: Plugin.cs (hunk 4 above)
Comment (tesu) [reply Plugin.cs:168]: I think this is silly. The reason why we have the loop in the HQ case is so we skip over the NQ items when figuring out the cheapest listing to undercut.
Putting this loop in the NQ case means we skip over the HQ items when deciding when to undercut, so if an item has a 1000g HQ listing and then a 2000g NQ listing, you would undercut the 2000g one. I think most people would want to undercut the cheapest listing regardless of its NQ or HQ if the item they have is NQ, so we don't want the loop in the first place (i.e. if we have an NQ item, we always undercut the cheapest item (i = 0), not the cheapest NQ item (i = whatever you end up from this loop))

File: Plugin.cs (hunk 7 above)
Comment (Looty) [reply Plugin.cs:216]: Yes
https://github.com/daemitus/YesAlready/blob/master/YesAlready/YesAlreadyPlugin.cs#L119

File: Configuration.cs (hunk 6 above)
Comment (tesu) [reply Configuration.cs:14]: Cool. Wasn't sure if the old config settings would get read incorrectly or something.

**Review by tesu — APPROVED** [review]

File: Plugin.cs (hunk 3 above)
Comment (tesu) [reply Plugin.cs:150]: Ok, I can release this in testing for now. If the offset still works for the next patch (5.56 I think?), then I will be confident enough to update the main release.
